widget/list.go

```diff
@@ -116,17 +123,23 @@ func (l *listRenderer) Layout(size fyne.Size) {
 
 	// Relayout What Is Visible - no scroll change - initial layout or possibly from a resize.
 	l.visibleItemCount = int(math.Ceil(float64(l.scroller.size.Height) / float64(l.list.itemMin.Height)))
+	if len(l.children) > l.list.Length() {
```

Comment:
I did not notice that this code was repeated 3 times sorry, I can't help but think this is not all needed.
For example the first two are doing the same, reducing to the smaller of l.list.Length() and l.visibleItemCount (I think).
So this could be a single loop where the end point was the smaller of these two?